packages/flutter_tools/README.md

```diff
@@ -1,13 +1,58 @@
 # Flutter Tools
 
-Developer tools for building Flutter applications.
+Command line developer tools for building Flutter applications.
 
-Be sure to follow the instructions on
-[CONTRIBUTING.md](../../CONTRIBUTING.md) to setup.
+## Working on Flutter Tools
 
-To run the tests, ensure that no devices are connected,
-then navigate to `flutter_tools` and execute:
+Be sure to follow the instructions on [CONTRIBUTING.md](../../CONTRIBUTING.md)
+to set up your development environment.
 
+### Setup
+
+First, ensure that the Dart SDK and other necessary artifacts are available by
+invoking the Flutter Tools wrapper script. In this directory run:
+```shell
+$ ../../bin/flutter --version
+```
+
+### Running
+
+To run Flutter Tools from source, in this directory run:
```

Comment:
Sorry, one more:

To re-generate the snapshotted tool, delete `bin/cache/flutter_tools.stamp` and `bin/cache/flutter_tools.snapshot`

Reply:
Added.